h2o-algos/src/main/java/hex/rulefit/Condition.java

```diff
@@ -104,4 +109,50 @@ public void map(Chunk[] cs, byte[] out) {
             out[iRow] = newVal;
         }
     }
+
+    Condition expandBy(Condition otherCondition) {
+        assert this.type.equals(otherCondition.type);
+        assert this.operator.equals(otherCondition.operator);
+        assert this.featureIndex == otherCondition.featureIndex;
+        assert this.featureName.equals(otherCondition.featureName);
```

Comment:
What about having a class field `condition` with all the information and implementing the `equals` method in the `Condition` class to make only one assertion here? :) 

```suggestion
        assert this.condition.equals(otherCondition);
```

Reply:
Here I am merging 2 conditions with different thresholds, like x < 4 and x <= 6.5 should result in x < 4, so this is not what I would expect from Condition.equals() method. I am expecting here that there will be a difference which I want to merge into one condition. In this part of code it shouldn't happen that  I will have here for example: x < 4 and z < 5 (difference in feture) or x > 5 and x < 8 (difference in operator) or x < 5 and y in {A, B, C} (difference in type and feature). Hence, I put only asserts here. I can implement Conditions.equals() to check equality of 2 conditions through all properties, but currently I don't need this anywhere.

Reply:
Oh, I see. Sure, then it is not necessary to create exactly the `equals` function. Maybe some kind of check function, which returns true if everything is ok, otherwise false. But it is up to you.